cluster-autoscaler/cloudprovider/packet/packet_manager_rest.go

```diff
@@ -180,33 +182,49 @@ func (mgr *packetManagerRest) listPacketDevices() (*Devices, error) {
 	var jsonStr = []byte(``)
 	packetAuthToken := os.Getenv("PACKET_AUTH_TOKEN")
 	url := mgr.baseURL + "/projects/" + mgr.projectID + "/devices"
-	req, _ := http.NewRequest("GET", url, bytes.NewBuffer(jsonStr))
+
+	req, err := http.NewRequest("GET", url, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return nil, fmt.Errorf("failed to list Packet devices: %w", err)
+	}
+
 	req.Header.Set("X-Auth-Token", packetAuthToken)
 	req.Header.Set("Content-Type", "application/json")
 	client := &http.Client{}
+
 	resp, err := client.Do(req)
 	if err != nil {
-		panic(err)
-		// klog.Fatalf("Error listing nodes: %v", err)
+		return nil, fmt.Errorf("failed to list Packet devices: %w", err)
 	}
+
 	defer resp.Body.Close()
 
 	klog.Infof("response Status: %s", resp.Status)
 
-	var devices Devices
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	if resp.StatusCode == http.StatusOK {
```

Comment:
I'd swap the logic and return error here, so we keep happy path aligned to left side.
```suggestion
	if resp.StatusCode != http.StatusOK {
```